packages/node_modules/@webex/test-users/src/index.js

```diff
@@ -0,0 +1,290 @@
+import assert from 'assert';
+
+import btoa from 'btoa';
+import _ from 'lodash';
+import randomName from 'node-random-name';
+import {request} from '@webex/http-core';
+
+import generateRandomString from './generate-random-string.js';
+
+
+const BASE_PATH_SECURE = '/users/test_users_s';
+const BASE_PATH = '/users/test_users';
+
+/**
+ * Computes `expires` and `refresh_token_expires` from `expires_in` and
+ * `refresh_token_expires_in` and creates an `authorization` string.
+ * @param {Object} token
+ * @private
+ * @returns {Object}
+ */
+function fixToken(token) {
+  const now = Date.now();
+
+  if (token.expires_in && !token.expires) {
+    token.expires = now + token.expires_in * 1000;
+  }
+
+  if (token.refresh_token_expires_in && !token.refresh_token_expires) {
+  /* eslint camelcase: [0] */
+    token.refresh_token_expires = now + token.refresh_token_expires_in * 1000;
+  }
+
+  if (token.token_type && token.access_token) {
+    token.authorization = `${token.token_type} ${token.access_token}`;
+  }
+
+  return token;
+}
+
+let clientToken;
+
+/**
+ * Fetches credentials to talk to the test_users_s endpoint
+ *
+ * Caches result in `clientToken` variable for multiple runs
+ * @param {Object} options
+ * @param {string} options.clientId
+ * @param {string} options.clientSecret
+ * @param {string} options.idbrokerUrl
+ * @private
+ * @returns {String}
+ */
+function getClientCredentials({clientId, clientSecret, idbrokerUrl}) {
+  if (clientToken) {
+    return Promise.resolve(clientToken);
+  }
+
+  return request({
+    method: 'POST',
+    uri: `${idbrokerUrl}/idb/oauth2/v1/access_token`,
+    json: true,
+    form: {
+      grant_type: 'client_credentials',
+      scope: 'webexsquare:get_conversation',
+      client_id: clientId,
+      client_secret: clientSecret
+    },
+    headers: {
+      // Note: we can't request's auth hash here because this endpoint expects
+      // us to send the auth header *without including "Basic "* before the
+      // token string
+      authorization: btoa(`${clientId}:${clientSecret}`)
+    }
+  })
```

Comment:
Great use of decoupling here to pull the params away from other resources. This allows the plugin to potentially work outside of all other plugins.

Reply:
Yep! That's what I was going for here. Since this is meant to be used outside of the actual "webex" ecosystem.